src/components/Navbar/SearchDrawer/SearchDrawer.tsx

```diff
@@ -151,13 +154,45 @@ const SearchDrawer: React.FC = () => {
           </div>
         </div>
       </div>
-      <div className={styles.bodyContainer}>
-        <SearchResults
-          searchResult={searchResult}
-          searchQuery={searchQuery}
-          isSearching={isSearching}
-          hasError={hasError}
-        />
+      <div
```

Comment:
nit: This is too much nesting for my liking, IMO it's better to have multiple return statements based on the condition. Instead of a huge return statement. 

Also, instead of 

```
 !searchQuery ||
            isSearching ||
            hasError ||
            (!isSearching &&
              !hasError &&
              searchResult &&
              searchResult.pagination.totalRecords === 0)
```

we can encapsulate this logic in a single variable (or a couple depending on what makes sense) 

ditto for the other statements